src/client/csfle/state_machine.rs

```diff
@@ -89,11 +107,21 @@ impl CryptExecutor {
                         Error::internal("db required for NeedMongoMarkings state")
                     })?;
                     let op = RawOutput(RunCommand::new_raw(db.to_string(), command, None, None)?);
-                    let mongocryptd_client = self
-                        .mongocryptd_client
+                    let mongocryptd = self
+                        .mongocryptd
                         .as_ref()
                         .ok_or_else(|| Error::internal("mongocryptd client not found"))?;
-                    let response = mongocryptd_client.execute_operation(op, None).await?;
+                    let mut result = mongocryptd.client.execute_operation(op.clone(), None).await;
+                    if let Err(e) = &result {
+                        if matches!(&*e.kind, ErrorKind::ServerSelection { .. }) {
+                            mongocryptd.respawn().await?;
```

Comment:
DRIVERS-1861 also mentions that we're supposed to be doing this spawning at MongoClient construction rather than on first operation, but given that it could technically involve SRV resolution (and therefore be async), I think we'll have to ignore that and keep it as we have it here. It's not encoded in the specification anyways, so we're not technically violating anything by doing so.

Reply:
Ah right, I forgot we had made a separate method for that.